### PR TITLE
Initial typescript config

### PR DESCRIPTION
**Pull request by Meachamp**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    "parser": "@typescript-eslint/parser",
+    "plugins": [
+      "@typescript-eslint"
+    ],
+    "extends": [
+      "eslint:recommended",
+      "plugin:@typescript-eslint/eslint-recommended",
+      "plugin:@typescript-eslint/recommended"
+    ]
+  }
```
Comment (ChristopherJMiller) [inline .eslintrc:12]: Please configure eslint with the rule to add a new line to the end of files.

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
```
Comment (ChristopherJMiller) [inline tsconfig.json:7]: Please bump to es2020 or esNext (I believe the module is the only thing confined for functionality with NodeJS)

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    // "mapRoot": "",                         /* Specify the location where debugger should locate map files instead of generated locations. */
+    // "inlineSourceMap": true,               /* Emit a single file with source maps instead of having a separate file. */
+    // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */
+
+    /* Experimental Options */
+    // "experimentalDecorators": true,        /* Enables experimental support for ES7 decorators. */
+    // "emitDecoratorMetadata": true,         /* Enables experimental support for emitting type metadata for decorators. */
+
+    /* Advanced Options */
+    "skipLibCheck": true,                     /* Skip type checking of declaration files. */
```
Comment (ChristopherJMiller) [inline tsconfig.json:67]: I wouldn't recommend this flag. Did you set this for any particular package?

**Review by ChristopherJMiller — CHANGES_REQUESTED** [review]
Pretty good setup overall, a couple issues and questions with the typescript config file.

File: tsconfig.json (hunk 3 above)
Comment (Meachamp) [reply tsconfig.json]: Unsure why this was generated by typescript init. I've looked up what it does and removed it. Will add back if there are any issues with library compatibility.

**Review by ChristopherJMiller — APPROVED** [review]
LGTM

**Pull request merged**